src/http.ts

```diff
@@ -80,12 +83,9 @@ export class Http {
 		return new Http(config);
 	}
 
-	errorTransform(method: Function) {
-		const http = new Http(this.configuration);
-		if (method) {
-			http.errorTransformMethod = method;
-		}
-		return http;
+	errorTransform<TIn, TOut>(method: ErrorTransform<TIn, TOut>) {
+		const config = { ...this.configuration, errorTransform: method };
+		return new Http(config);
 	}
```

Comment:
This was problematic because it's setting the transform function as an instance variable that won't get passed on to subsequent instances if another method is chained. Moving it into the `HttpConfig` object ensures it gets passed along.